src/drivers/display/renesas_glcdc/lv_renesas_glcdc.c

```diff
@@ -0,0 +1,192 @@
+{
+    glcdc_init();
+
+    lv_display_t * display = lv_display_create(DISPLAY_HSIZE_INPUT0, DISPLAY_VSIZE_INPUT0);
+    lv_display_set_flush_cb(display, flush_partial);
+    lv_display_set_flush_wait_cb(display, flush_wait_partial);
+    lv_display_set_buffers(display, buf1, buf2, buf_size, LV_DISPLAY_RENDER_MODE_PARTIAL);
+
+    return display;
+}
+
+/*This function is declared in and being used by FSP generated code modules*/
+void glcdc_callback(display_callback_args_t * p_args)
+{
+    if(DISPLAY_EVENT_LINE_DETECTION == p_args->event) {
+#if BSP_CFG_RTOS == 2               /*FreeRTOS*/
+        BaseType_t context_switch;
+
+        /*Set Vsync semaphore*/
+        xSemaphoreGiveFromISR(_SemaphoreVsync, &context_switch);
+
+        /*Return to the highest priority available task*/
+        portYIELD_FROM_ISR(context_switch);
+#else
+#endif
+    }
+    else if(DISPLAY_EVENT_GR1_UNDERFLOW == p_args->event) {
+        __BKPT(0); /*Layer 1 Underrun*/
+    }
+    else if(DISPLAY_EVENT_GR2_UNDERFLOW == p_args->event) {
+        __BKPT(0); /*Layer 2 Underrun*/
+    }
+    else { /*DISPLAY_EVENT_FRAME_END*/
+        __BKPT(0);
+    }
+
+}
+
+/**********************
+ *   STATIC FUNCTIONS
+ **********************/
+
+static void glcdc_init(void)
+{
+    /* Fill the Frame buffer with black colour (0x0000 in RGB565), for a clean start after previous runs */
+    lv_memzero(fb_background, sizeof(fb_background));
+
+    /* Initalize GLCDC driver */
+    uint8_t * p_fb = &fb_background[1][0];
+    fsp_err_t err;
+
+    err = R_GLCDC_Open(&g_display0_ctrl, &g_display0_cfg);
+    if(FSP_SUCCESS != err) {
+        __BKPT(0);
+    }
+
+    err = R_GLCDC_Start(&g_display0_ctrl);
+    if(FSP_SUCCESS != err) {
+        __BKPT(0);
+    }
+
+    do {
+        err =
+            R_GLCDC_BufferChange(&g_display0_ctrl,
+                                 (uint8_t *) p_fb,
+                                 (display_frame_layer_t) 0);
+    } while(FSP_ERR_INVALID_UPDATE_TIMING == err);
+}
+
+static void flush_direct(lv_display_t * display, const lv_area_t * area, uint8_t * px_map)
+{
+    FSP_PARAMETER_NOT_USED(area);
+    /*Display the frame buffer pointed by px_map*/
+
+    if(!lv_display_flush_is_last(display)) return;
+
+#if defined(RENESAS_CORTEX_M85) && (BSP_CFG_DCACHE_ENABLED)
+    /* Invalidate cache - so the HW can access any data written by the CPU */
+    SCB_CleanInvalidateDCache_by_Addr(px_map, sizeof(fb_background[0]));
+#endif
+
+    R_GLCDC_BufferChange(&g_display0_ctrl,
+                         (uint8_t *) px_map,
+                         (display_frame_layer_t) 0);
+}
+
+static void flush_wait_direct(lv_display_t * display)
+{
+    if(!lv_display_flush_is_last(display)) return;
+
+#if BSP_CFG_RTOS == 2              /*FreeRTOS*/
+    /*If Vsync semaphore has already been set, clear it then wait to avoid tearing*/
+    if(uxSemaphoreGetCount(_SemaphoreVsync)) {
+        xSemaphoreTake(_SemaphoreVsync, 10);
```

Comment:
Based on your comment, should this line give/clear the semaphore instead of take it?

Reply:
It was copied from the original code, and it worked, but I don't understand something. 

The idea is to take the semaphore twice:
1. Take it normally (nothing happens)
2. Take it again to yield and wait until [VSYNC gives it](https://github.com/lvgl/lvgl/pull/6221/files/1563b8df398e10cf5d6dca371a37dc62285ed7d0#diff-faec9062bf810d2b190e1ce87e4165dfa0b63423d074de2645fb851e50b64fddR79)

But I can't see how `if(uxSemaphoreGetCount(_SemaphoreVsync))` can be `true` as the semaphore is is always given at the end. And if it's not taken twice this mechanism is no t working.  

@becseya can you take a deeper look on this part? What happens if you just add
```c
xSemaphoreTake(_SemaphoreVsync, portMAX_DELAY);
xSemaphoreTake(_SemaphoreVsync, portMAX_DELAY);
```
here?

Reply:
kisvegabor's comment helped me understand how this logic works. I think it's actually okay. The code comment wording confused me because `set` means `give` and `clear` means `take`, in the code comment.

Also, regarding
> But I can't see how `if(uxSemaphoreGetCount(_SemaphoreVsync))` can be `true` as the semaphore is is always given at the end.

`GetCount` returns 1 if it's _available_ [according to the docs](https://www.freertos.org/uxSemaphoreGetCount.html)

So it should work as a signalling mechanism, as intended. I'm going to approve.